hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/feedcomm/ExportJdrCommand.java

```diff
@@ -0,0 +1,152 @@
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Collection;
+
+import org.hawkular.agent.monitor.extension.MonitorServiceConfiguration;
+import org.hawkular.agent.monitor.inventory.ID;
+import org.hawkular.agent.monitor.inventory.InventoryIdUtil;
+import org.hawkular.agent.monitor.inventory.ManagedServer;
+import org.hawkular.agent.monitor.inventory.Name;
+import org.hawkular.agent.monitor.inventory.ResourceManager;
+import org.hawkular.agent.monitor.inventory.dmr.DMRInventoryManager;
+import org.hawkular.agent.monitor.inventory.dmr.DMROperation;
+import org.hawkular.agent.monitor.inventory.dmr.DMRResource;
+import org.hawkular.agent.monitor.inventory.dmr.LocalDMRManagedServer;
+import org.hawkular.agent.monitor.inventory.dmr.RemoteDMRManagedServer;
+import org.hawkular.agent.monitor.log.MsgLogger;
+import org.hawkular.bus.common.BasicMessageWithExtraData;
+import org.hawkular.bus.common.BinaryData;
+import org.hawkular.cmdgw.api.ExportJdrRequest;
+import org.hawkular.cmdgw.api.ExportJdrResponse;
+import org.hawkular.dmrclient.Address;
+import org.hawkular.dmrclient.CoreJBossASClient;
+import org.hawkular.dmrclient.JBossASClient;
+import org.hawkular.inventory.api.model.CanonicalPath;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.dmr.ModelNode;
+
+/**
+ * @author Juraci Paixão Kröhling
+ */
+public class ExportJdrCommand implements Command<ExportJdrRequest, ExportJdrResponse> {
+    public static final Class<ExportJdrRequest> REQUEST_CLASS = ExportJdrRequest.class;
+
+    @Override
+    public BasicMessageWithExtraData<ExportJdrResponse> execute(ExportJdrRequest request, BinaryData binaryData,
+                                                                CommandContext context)
+            throws Exception {
+        MsgLogger.LOG.infof("Received request to generate JDR on resource [%s]", request.getResourcePath());
+
+        FeedCommProcessor processor = context.getFeedCommProcessor();
+        MonitorServiceConfiguration config = context.getMonitorServiceConfiguration();
+
+        // Based on the resource ID we need to know which inventory manager is handling it.
+        // From the inventory manager, we can get the actual resource.
+        CanonicalPath canonicalPath = CanonicalPath.fromString(request.getResourcePath());
+        String resourceId = canonicalPath.ids().getResourcePath().getSegment().getElementId();
+        InventoryIdUtil.ResourceIdParts idParts = InventoryIdUtil.parseResourceId(resourceId);
+        ManagedServer managedServer = config.managedServersMap.get(new Name(idParts.getManagedServerName()));
+        if (managedServer == null) {
+            throw new IllegalArgumentException(String.format("Cannot execute operation: unknown managed server [%s]",
+                    idParts.getManagedServerName()));
+        }
+
+        if (managedServer instanceof LocalDMRManagedServer || managedServer instanceof RemoteDMRManagedServer) {
```

Comment:
It is today, but not in the future. This is just code for the future - later on when we introduce JMS-based managed servers, we'll want to do something different for those.